samples/snippets/src/main/java/com/example/bigquerystorage/WriteCommittedStream.java

```diff
@@ -45,54 +57,104 @@ public static void runWriteCommittedStream()
-        client.finalizeWriteStream(finalizeWriteStreamRequest);
+        writer.append(jsonArr, i * 10);
       }
-      System.out.println("Appended records successfully.");
     } catch (ExecutionException e) {
       // If the wrapped exception is a StatusRuntimeException, check the state of the operation.
       // If the state is INTERNAL, CANCELLED, or ABORTED, you can retry. For more information, see:
       // https://grpc.github.io/grpc-java/javadoc/io/grpc/StatusRuntimeException.html
-      System.out.println("Failed to append records. \n" + e.toString());
+      System.out.println("Failed to append records. \n" + e);
+    }
+
+    // Final cleanup.
+    writer.cleanup();
+    System.out.println("Appended records successfully.");
+  }
+
+  public static WriteCommittedStream create(String projectId, String datasetName, String tableName)
+      throws IOException, DescriptorValidationException, InterruptedException {
+    WriteCommittedStream writer = new WriteCommittedStream();
+    writer.client = BigQueryWriteClient.create();
+    // Initialize a write stream for the specified table.
+    // For more information on WriteStream.Type, see:
+    // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1beta2/WriteStream.Type.html
+    WriteStream stream = WriteStream.newBuilder().setType(WriteStream.Type.COMMITTED).build();
+    TableName parentTable = TableName.of(projectId, datasetName, tableName);
+    CreateWriteStreamRequest createWriteStreamRequest =
+        CreateWriteStreamRequest.newBuilder()
+            .setParent(parentTable.toString())
+            .setWriteStream(stream)
+            .build();
+    writer.writeStream = writer.client.createWriteStream(createWriteStreamRequest);
+
+    // Use the JSON stream writer to send records in JSON format.
+    // For more information about JsonStreamWriter, see:
+    // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.html
+    writer.streamWriter =
+        JsonStreamWriter.newBuilder(
+                writer.writeStream.getName(), writer.writeStream.getTableSchema())
+            .build();
+    return writer;
+  }
+
+  public void append(JSONArray data, int offset)
+      throws DescriptorValidationException, IOException, ExecutionException, InterruptedException {
+    // To detect duplicate records, pass the index as the record offset.
+    // To disable deduplication, omit the offset or use WriteStream.Type.DEFAULT.
+    ApiFuture<AppendRowsResponse> future = streamWriter.append(data, /*offset=*/ offset);
+    this.inflightRequestPhaser.register();
+    ApiFutures.addCallback(
+        future, new AppendCompleteCallback(this), MoreExecutors.directExecutor());
+  }
+
+  public void cleanup() {
+    // Wait for all in-flight requests to complete.
+    this.inflightRequestPhaser.arriveAndAwaitAdvance();
```

Comment:
Close the JsonWriter here? Since you are doing a Finalize.

Reply:
Added the close, but as we discussed offline, we still need to wait for inflight requests to finish